python/instrumentation/openinference-instrumentation-langchain/src/openinference/instrumentation/langchain/__init__.py

```diff
@@ -64,6 +64,20 @@ def _uninstrument(self, **kwargs: Any) -> None:
     def get_span(self, run_id: UUID) -> Optional[Span]:
         return self._tracer.get_span(run_id) if self._tracer else None
 
+    def get_root_chain_span(self, run_id: UUID) -> Optional[Span]:
```

Comment:
I think by making the following signature change, we could make this more general and at the same time simplify our internal logic, given the following compositional equivalence. This simplifies our backend logic because we only need a `Dict[UUID, Optional[UUID]]` to track whose parent is whom, and we can just call on the preexisting `self.run_map`.
```python
get_root_chain_span(run_id) is get_span(get_root_chain_run_id(run_id))
```
We just have to also update `get_span` to take `None` as argument, which is a trivial change.
```suggestion
    def get_root_chain_run_id(self, run_id: UUID) -> Optional[UUID]:
```

Reply:
FYI, [`self.run_map`](https://github.com/langchain-ai/langchain/blob/2a1029c53c9c14b177eb61743dc0501f97dd0738/libs/core/langchain_core/tracers/core.py#L117) seems to be all you need.